src/charm.py

```diff
@@ -125,13 +112,11 @@ def __init__(self, *args):
         )
         self.framework.observe(self.on.update_status, self._on_update_status)
         self.framework.observe(self.tracing.on.request, self._on_tracing_request)
-        self.framework.observe(self.on.tracing_relation_created, self._on_tracing_relation_created)
-        self.framework.observe(self.on.tracing_relation_joined, self._on_tracing_relation_joined)
-        self.framework.observe(self.on.tracing_relation_changed, self._on_tracing_relation_changed)
         self.framework.observe(self.on.collect_unit_status, self._on_collect_unit_status)
         self.framework.observe(self.on.list_receivers_action, self._on_list_receivers_action)
         self.framework.observe(self.cert_handler.on.cert_changed, self._on_cert_handler_changed)
         self.framework.observe(self.on.config_changed, self._on_config_changed)
+        self.framework.observe(self.on.tracing_relation_broken, self._on_tracing_relation_broken)
```

Comment:
I thought we'd agree to keep this in the tracing lib and add a new custom event 
```suggestion
        self.framework.observe(self.tracing.on.broken, self._on_tracing_broken)
```